TrafficSimulator/Assets/Utilities/CarSpawner.cs

```diff
@@ -142,32 +148,87 @@ private void SpawnCars()
             {
                 // Calculate the number of cars to spawn
                 int carsToSpawn = _mode == SpawnMode.Total ? Mathf.CeilToInt(_ratios[i] * TotalCars) : Mathf.CeilToInt(_maxCarsPerLane[i] * LaneCarRatio);
-
                 // Return if there are no cars to spawn
                 if (carsToSpawn == 0)
                     return;
 
                 // Calculate the offset
                 _laneNodeCurrent = _lanes[i].StartNode;
 
-                // Spawn cars
-                for (int j = 0; j < carsToSpawn; j++)
+                List<float> sections = DivideLaneToSections(_lanes[i]);
+                List<float> sectionRatios = CalculateSectionRatios(sections);
+                for (int j = 0; j < sections.Count; j++)
                 {
-                    // Check if max cars have been spawned
-                    if (_mode == SpawnMode.Total && _carCounter >= TotalCars)
-                        return;
+                    // Calculate the number of cars to spawn
+                    int carsToSpawnSection = Mathf.CeilToInt(carsToSpawn * sectionRatios[j]);
+                    spawnCarsInSection(_lanes[i], i, carsToSpawnSection, sections[j]);
+                }
+                _offset = 0;
+            }
+        }
 
-                    // Spawn individual car at current node
-                    SpawnCar(i);
+        /// <summary>Spawns cars in a section</summary>
+        private void spawnCarsInSection(Lane lane, int laneIndex, int carsToSpawn, float sectionLength)
+        {
+            for (int i = 0; i < carsToSpawn; i++)
+            {
+                // Check if max cars have been spawned
+                if (_mode == SpawnMode.Total && _carCounter >= TotalCars)
+                    return;
 
+                // Spawn car
+                if (isCarSpawnable(_laneNodeCurrent))
+                {
+                    SpawnCar(laneIndex);
                     _carCounter++;
-                    _offset += _lanes[i].Length / carsToSpawn;
-                    _laneNodeCurrent = CalculateSpawnNode(_offset, _lanes[i]);
                 }
-                _offset = 0;
+                // Calculate the next spawn node in the section based on distance
+                _offset += sectionLength / (carsToSpawn);
+                _laneNodeCurrent = CalculateSpawnNode(_offset, lane);
             }
         }
 
+        /// <summary>Divides a lane into sections based on intersections</summary>
+        private List<float> DivideLaneToSections(Lane lane)
+        {
+            LaneNode curr = lane.StartNode;
+            LaneNode prev;
+            List<float> sections = new List<float>();
+            float sectionLength = 0;
+
+            while (curr != null)
```

Comment:
The code in lane is more readable and easier to understand, you could copy that code and add a new section in the if(isIntersection) statement